cmd/cluster/core/create.go

```diff
@@ -164,14 +168,25 @@ func createCommonFixture(ctx context.Context, opts *CreateOptions) (*apifixtures
 		k, v := pair[0], pair[1]
 		annotations[k] = v
 	}
+	// Add annotation to support import into MCE
+	annotations[managedClusterNameKey] = opts.Name
 
 	if len(opts.ControlPlaneOperatorImage) > 0 {
 		annotations[hyperv1.ControlPlaneOperatorImageAnnotation] = opts.ControlPlaneOperatorImage
 	}
 
-	pullSecret, err := ioutil.ReadFile(opts.PullSecretFile)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read pull secret file: %w", err)
+	var pullSecret []byte
+	var err error
+	if util.IsEmpty(opts.PullSecretFile) && util.IsNotEmpty(opts.CredentialSecretName) {
```

Comment:
This allows the --pull-secret parameter to override the secret, otherwise use the secret data.

Reply:
why do we want that?
can we just simplify and if CredentialSecretName use that and ignore everything else?

Reply:
If we still want the fallback I'd avoid the if else, e.g.
```
	if opts.CredentialSecretName != "" {
		pullSecret, err = util.GetDockerConfigJSON(opts.CredentialSecretName, opts.Namespace)
		if err != nil {
			return nil, err
		}
	} 
	
        // overrides secret if set
	if opts.PullSecretFile != "" {
		pullSecret, err = ioutil.ReadFile(opts.PullSecretFile)
		if err != nil {
			return nil, fmt.Errorf("failed to read pull secret file: %w", err)
		}
	}
```

Reply:
I would have preferred not to do the fallback and just always use the credential, but I got the feedback from the team and @derekwaynecarr that on the CLI folks might want to mix and match.

I would rather the simple flow with no overrides. Just do a check to see if they specified --base-domain, pull-secret or sshkey with --secret-creds and if that is the case, throw an error.

Reply:
Suggestion implemented and still supports overrides